src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -221,7 +221,10 @@ namespace xrt {
       }
 
       ffi_call(&cif,FFI_FN(kernel), &kernel_return, ffi_arg_values);
-      args_from_host[return_offset] = (uint32_t)kernel_return;
+      if (kernel_return >= 0)
```

Comment:
My I suggest the following:
`args_from_host[return_offset] = (kernel_return >= 0) ? static_cast<uint32_t>(kernel_return) : 255;`